File: jobboard-frontend/src/services/aiService.js
import axios from "../api/axiosConfig";

const generateJobDescription = (title) => {
    return axios.post(
        "/ai/generate-description",
        title,
        {
            headers: {
                "Content-Type": "text/plain"
            }
        }
    );
};

const analyzeResume = (jobDescription, resume) => {
    return axios.post(
        "/ai/analyze-resume",
        {
            jobDescription: jobDescription,
            resume: resume
        }
    );
};

export default {
    generateJobDescription,
    analyzeResume
};

File: jobboard-backend/src/main/java/com/amar/jobboard/ai/service/AiService.java
package com.amar.jobboard.ai.service;

import com.amar.jobboard.ai.dto.ResumeAnalysisResponse;

public interface AiService {
	String generateJobDescription(String title);
	ResumeAnalysisResponse analyzeResume(
            String jobDescription,
            String resume
    );

}
